spec/Appendix B -- Variable Batching.md

```diff
@@ -0,0 +1,170 @@
+
+```graphql
+query($ids: [ID!]!) {
+    productsByIds(ids: $ids) {
+        id
+        name
+    }
+}
+```
+
+However, this approach doesn’t work when variables are used within nested selections.
+
+```graphql
+query($ids: [ID!]! $first: Int!) {
+    productsByIds(ids: $ids) {
+        id
+        name
+        reviews(first: $first) {
+            id
+            body
+        }
+    }
+}
+```
+
+Variable batching allows a single request to include multiple sets of variables. This approach is particularly useful for GraphQL gateways to fetch multiple entities by their keys and bulk operations, reducing network overhead and improving performance by consolidating multiple operations into a single request.
+
+### Variable Batching Request
+
+A server MAY accept a **variable batching request** via `POST`.
+
+#### Request Parameters
+
+A _variable batching request_ follows the same structure as a standard GraphQL request, with the key difference being in the `variables` parameter. All other parameters are consistent with the standard GraphQL-over-HTTP protocol:
+
+- {query} - (_Required_, string): The string representation of the Source Text
+  of a GraphQL Document as specified in
+  [the Language section of the GraphQL specification](https://spec.graphql.org/draft/#sec-Language).
+- {operationName} - (_Optional_, string): The name of the Operation in the
+  Document to execute.
+- **{variables}** (Required, array of maps):
+  Instead of a single map, the `variables` parameter is an array of maps. Each map in the array represents a different set of variables to be used in executing the operation.
+- {extensions} - (_Optional_, map): This entry is reserved for implementors to
+  extend the protocol however they see fit.
+
+### Accept
+
+A client SHOULD indicate the media types that it supports in responses using the `Accept` HTTP header as specified in [RFC7231](https://datatracker.ietf.org/doc/html/rfc7231).
+
+For **variable batching requests**, the client SHOULD include the media type `application/graphql+jsonl` in the `Accept` header to indicate that it expects a batched response in JSON Lines format.
+
+If the client supplies an Accept header, the client SHOULD include the media type `application/graphql+jsonl` in the Accept header.
+
+If the client does not supply an `Accept` header, the server MAY respond with a default content type of `application/graphql+jsonl`, or it may use any other media type it supports. However, to ensure compatibility and clarity, it is RECOMMENDED that the client explicitly states its preferred media types using the `Accept` header.
+
+### POST
+
+A **variable batching request** instructs the server to perform a query or mutation operation multiple times, once for each set of variables provided. The request MUST have a body that contains the values of the _variable batching request_ parameters encoded in the `application/graphql+jsonl` media type, or another media type supported by the server.
```

Comment:
We wouldn't encode the parameters in JSONL? Just a simple JSON value is fine - they're all sent at the same time.